### PR TITLE
dist: added ng_sniffer script

### PR DESCRIPTION
**Pull request by haukepetersen**
I rewrote the sniffer script a little bit to work with the new network stack. 

The script works together with the application proposed in https://github.com/RIOT-OS/applications/pull/6

### CODE REVIEW

**Comment by authmillenon** [comment]
What exactly did you change and why was it necessary?

**Comment by OlegHahm** [comment]
I think it's mostly that the syntax for setting channels and activating promiscuous mode has changed in the gnrc stack.

**Comment by haukepetersen** [comment]
that and also some code simplification and cleanup

**Comment by OlegHahm** [comment]
Actually, now that I think of it, we could actually merge both scripts. But since the older one is supposed to vanish (fingers crossed), we could save this work.

File: dist/tools/ng_sniffer/ng_sniffer.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        sys.exit(2)
+
+time.sleep(1)
+
+sys.stderr.write('ifconfig 5 set chan %s\n' % sys.argv[3])
+sys.stderr.write('ifconfig 5 set raw 1\n')
+sys.stderr.write('ifconfig 5 set promiscuous 1\n')
+serport.write('ifconfig 5 set raw 1\n')
+serport.write('ifconfig 5 set chan %s\n' % sys.argv[3])
+serport.write('ifconfig 5 set promiscuous 1\n')
```
Comment (authmillenon) [inline dist/tools/ng_sniffer/ng_sniffer.py]: With modifications in https://github.com/RIOT-OS/applications/pull/6#issuecomment-105901087 this is interface 4 now

**Comment by authmillenon** [comment]
A README, that this is to be used with RIOT-OS/applications#6 would be nice.

**Comment by haukepetersen** [comment]
updated the script to the new semantics of the `ifconfig` shell command and added a `README`. Say when I should squash.

**Comment by haukepetersen** [comment]
@OlegHahm, @authmillenon, should I squash?

File: dist/tools/ng_sniffer/README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# RIOT Sniffer Application
+
+
+## About
+
+This sniffer application can be used to sniff network traffic using RIOT based
```
Comment (authmillenon) [inline dist/tools/ng_sniffer/README.md]: This is not the application, but a script to use the application ;-)

File: dist/tools/ng_sniffer/README.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+2. Run the `ng_sniffer.py` script
+```
+$ ./ng_sniffer.py <tty> <baudrate> <channel> [outfile]
+```
+
+### Examples
+
+The following examples are made when using the sniffer application together with
+an `iot-lab_M3` node that is connected to /dev/ttyUSB1 (or COM1) and runs per
+default with a baudrate of 500000.
```
Comment (authmillenon) [inline dist/tools/ng_sniffer/README.md]: Is the baudrate the same for all boards, or may there be differences?

Comment (haukepetersen) [inline dist/tools/ng_sniffer/README.md]: what do you mean? All iot-labs are per dafault configured to 500k baud. Most other boards in RIOT are configured to 115200.

Comment (authmillenon) [inline dist/tools/ng_sniffer/README.md]: Yeah, but we support more than just the IoT-Lab nodes. Or don't we support them for the sniffer? In theory every board supporting the at86rf2xx driver should be able to work with the sniffer app, right?

Comment (haukepetersen) [inline dist/tools/ng_sniffer/README.md]: Then read the text: 'EXAMPLE' :-)

Comment (haukepetersen) [inline dist/tools/ng_sniffer/README.md]: And all other nodes are working just fine, that what the baudrate parameter is for.

Comment (authmillenon) [inline dist/tools/ng_sniffer/README.md]: All I'm saying is that it would be nice if this was mentioned here somewhere ;-)

Comment (authmillenon) [inline dist/tools/ng_sniffer/README.md]: See it like this: some people don't even know what a baudrate is or what the parameter is doing and might want to use the wireshark sniffer regardless.

Comment (haukepetersen) [inline dist/tools/ng_sniffer/README.md]: Really? If you don't know what a baudrate is, start to learn something about computers and then come back to RIOT!

Comment (authmillenon) [inline dist/tools/ng_sniffer/README.md]: That's not very friendly…

Comment (haukepetersen) [inline dist/tools/ng_sniffer/README.md]: sorry :-)

**Comment by haukepetersen** [comment]
@authmillenon: how about this description of the scripts parameters?

File: dist/tools/ng_sniffer/README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+General usage:
+
+1. Flash an applicable RIOT node with the sniffer application from
+(https://github.com/RIOT-OS/applications/tree/master/ng_sniffer)
+
+2. Run the `ng_sniffer.py` script
+```
+$ ./ng_sniffer.py <tty> <baudrate> <channel> [outfile]
+```
+The script needs the following parameters:
```
Comment (authmillenon) [inline dist/tools/ng_sniffer/README.md]: "has" instead of "needs". `outfile` seems to be optional.

**Comment by authmillenon** [comment]
ACK, please squash.

**Comment by haukepetersen** [comment]
squashed.

**Comment by authmillenon** [comment]
Waiting for travis

**Comment by authmillenon** [comment]
Static tests are passing, the others should not be touched by this: Go.

**Pull request merged**